Spanner/src/TransactionalReadTrait.php

```diff
@@ -358,14 +362,18 @@ public function read($table, KeySet $keySet, array $columns, array $options = []
             $options['requestOptions']['transactionTag'] = $this->tag;
         }
 
-        return $this->operation->read(
+        $result = $this->operation->read(
             $this->session,
             $table,
             $keySet,
             $columns,
-            $options,
-            $this
+            $options
         );
+        $result->createGenerator();
```

Comment:
In the context of Database also creation of Generator should be fine. I was a bit skeptical about calling the `createGenerator` function in a constructor, thats why I placed it here. Since it can be a Global behavior, we can unconditionally call this method in the `Result::__construct` then.